### ISSUE TITLE
Calling the search endpoint with track as the only type returns albums

### ISSUE THREAD

**Issue by chan135** [spotify/web-api#504]
Issue found on February 19th 2015.

#### Endpoint(s):
GET https://api.spotify.com/v1/search

#### Scope(s):
* None (application is not using authentication)

#### Steps to reproduce:
1. Open the API Console
2. Call the GET https://api.spotify.com/v1/search

#### Expected behaviour:
JSON response contains tracks 

#### Actual behaviour:
JSON response returns albums

I need search to return just a list of tracks as I need to parse out the id of song, artist, and album art

**Comment by jscholes** [comment]
I can't reproduce this issue in the API console.  I'm using the search endpoint in my app also, and haven't ever seen this behaviour.  When you submit the request, the console will give you a curl command for executing the request on the command line.  Can you paste that here?  Make sure to remove your access token.

**Comment by chan135** [comment]
![1](https://cloud.githubusercontent.com/assets/14017301/25411963/d5d3935e-29ed-11e7-8e0a-5755fedbd113.PNG)
![2](https://cloud.githubusercontent.com/assets/14017301/25411961/d5d010bc-29ed-11e7-8a8c-89168eb8cb92.PNG)
![kendrick](https://cloud.githubusercontent.com/assets/14017301/25411964/d5d7787a-29ed-11e7-8889-3a5f7b5c9bdf.PNG)
![response](https://cloud.githubusercontent.com/assets/14017301/25411962/d5d0856a-29ed-11e7-9e29-30094ac7bf29.PNG)

**Comment by jscholes** [comment]
The request pictured above *does* only return tracks, or to be more specific, it returns a list of track objects which you can read more about in the [Spotify Web API Object Model](https://developer.spotify.com/web-api/object-model/#track-object).  As you'll see from those docs, all track objects include a simplified album object, but the actual search results are tracks, not albums.  If you set the search type to "album", you'll actually see a list of simplified album objects and no track data at all.

In other words,  the data you're after (i.e. artist and track URIs plus album art) is all there and nothing is amiss.  Track objects include their associated simplified album object as a convenience.

**Comment by chan135** [comment]
Ok, cool if I wanted to get the song ID as well as the artist ID and album art, how would I go around parsing that in JSON?

**Comment by jscholes** [comment]
- Every track object in the result set has an "id" key, as well as the "uri" key which includes the full Spotify URI prefix e.g. "spotify:track:" before the ID.
- The "artist" key holds the artist data for the track; this is an array of artists and each one will have both "id" and "uri" keys.
- As I described above, each track object also has an "album" key, which in turn has a key called "images".  Again this is an array - each image object has "height", "width" and "url" keys.

All this is covered in the Object Model documentation I linked to in my previous comment, but if you're having trouble working with the JSON yourself there are probably client libraries for your programming language of choice which can help you access the web API.  If I remember rightly you're using Python so you could give [Spotipy](https://spotipy.readthedocs.io/) a go.  There are plenty of tutorials out there to help you get started with JSON plus Python also, but in general terms once you've loaded the JSON response from the API you can work with it just like a Python dictionary and the object arrays become standard Python lists.  Hope that helps!

**Comment by chan135** [comment]
Ok, awesome thanks for the help!
